### DISCUSSION TITLE
/home/vscode/.profile is not sourced

### DISCUSSION THREAD

**Discussion by davetapley** [community/community#7919] [Codespaces] [answered]
Am I correct that `/home/vscode/.profile` is **not** sourced when starting a Codespace?

I am seeing this because I followed [these instructions](https://github.com/pyenv/pyenv/tree/df7ccfe0e89a6e7026518c992bf6ee107c6df26a#basic-github-checkout) to install `pyenv` in `mcr.microsoft.com/vscode/devcontainers/base:0-focal`, it works correctly when running in a devcontainer against a local Docker host, but fails in a Codespace (because `PATH` does get `$PYENV_ROOT/bin` added to it) 😞 

---

See also https://github.com/microsoft/vscode-dev-containers/issues/533 and https://github.com/pyenv/pyenv-installer/issues/112#issuecomment-974330744

**Comment by asciimike** [comment, marked as answer]
Per [our docs](https://docs.github.com/en/codespaces/customizing-your-codespace/personalizing-codespaces-for-your-account#dotfiles), I think we only check `~/.*` not `~/**/.*`. That said, it feels like we should probably load the vscode dotfiles :D 

You can probably run [a lifecycle script](https://code.visualstudio.com/docs/remote/devcontainerjson-reference#_lifecycle-scripts) to source it before the container is booted (e.g. running `onCreateCommand` or `initializeCommand`).
